lib/rules/no-async-in-computed-properties.js

```diff
@@ -0,0 +1,178 @@
+  'setImmediate',
+  'requestAnimationFrame'
+]
+
+function isTimedFunction (node) {
+  return (
+    node.callee.type === 'Identifier' &&
+    TIMED_FUNCTIONS.indexOf(node.callee.name) !== -1
+  ) || (
+    node.callee.type === 'MemberExpression' &&
+    node.callee.object.name === 'window' && (
+      TIMED_FUNCTIONS.indexOf(node.callee.property.name) !== -1 ||
+      TIMED_FUNCTIONS.indexOf(node.callee.property.value) !== -1
+    )
+  ) && node.arguments.length
+}
+
+function isPromise (node) {
+  if (node.type === 'CallExpression' && node.callee && node.callee.type === 'MemberExpression') {
+    return ( // hello.PROMISE_FUNCTION()
+      node.callee.property &&
+      PROMISE_FUNCTIONS.indexOf(node.callee.property.name) !== -1
+    ) || ( // Promise.PROMISE_METHOD()
+      node.callee.object.type === 'Identifier' &&
+      node.callee.object.name === 'Promise' &&
+      PROMISE_METHODS.indexOf(node.callee.property.name) !== -1
+    ) || ( // somePromise.ANYTHING()
+      node.callee.object && isPromise(node.callee.object)
+    )
+  }
+  return false
+}
+
+function create (context) {
+  const forbiddenNodes = []
+
+  function onFunctionEnter (node) {
+    if (node.async) {
+      forbiddenNodes.push({
+        node: node,
+        type: 'async'
+      })
+    }
+    if (node.generator) {
+      forbiddenNodes.push({
+        node: node,
+        type: 'yield*'
+      })
+    }
+  }
+
+  return Object.assign({},
+    {
+      FunctionDeclaration: onFunctionEnter,
+
+      FunctionExpression: onFunctionEnter,
+
+      ArrowFunctionExpression: onFunctionEnter,
+
+      NewExpression (node) {
+        if (node.callee.name === 'Promise') {
+          forbiddenNodes.push({
+            node: node,
+            type: 'new'
+          })
+        }
+      },
+
+      CallExpression (node) {
+        if (isPromise(node)) {
+          forbiddenNodes.push({
+            node: node,
+            type: 'promise'
+          })
+        }
+        if (isTimedFunction(node)) {
+          forbiddenNodes.push({
+            node: node,
+            type: 'timed'
+          })
+        }
+      },
+
+      YieldExpression (node) {
+        // await nodes are YieldExpression's with babel-eslint < 7.0.0
+        forbiddenNodes.push({
+          node: node,
+          type: 'yield'
+        })
+      },
+
+      AwaitExpression (node) {
+        forbiddenNodes.push({
+          node: node,
+          type: 'await'
+        })
+      }
+    },
+    utils.executeOnVueInstance(context, properties => {
+      const computedProperties = utils.getComputedProperties(properties)
+
+      computedProperties.forEach(cp => {
+        forbiddenNodes.forEach(el => {
+          if (
+            cp.value &&
+            el.node.loc.start.line >= cp.value.loc.start.line &&
+            el.node.loc.end.line <= cp.value.loc.end.line
+          ) {
+            let message = `Unexpected asynchronous action in "{{name}}" computed property.`
+            if (el.type === 'await') {
```

Comment:
I see a lot of repetition here. Please think about creating a map with all types and corresponding names. Then return the message without repeating it :)